encoding/json/decode.go

```diff
@@ -1266,6 +1279,19 @@ func (d *Decoder) decodeType(valueJSON any, results typeDecodingResults) cadence
 			d.gauge,
 			d.decodeType(obj.Get(typeKey), results),
 		)
+	case "Restriction":
+		// Backwards-compatibility for format <v1.0.0:
+		if d.backwardsCompatible {
+			restrictionsValue := obj.Get(restrictionsKey)
+			typeValue := obj.Get(typeKey)
+			return d.decodeDeprecatedRestrictedType(
+				typeValue,
+				toSlice(restrictionsValue),
+				results,
+			)
+		} else {
+			panic("Restriction kind is not supported")
+		}
```

Comment:
Bailout early, to avoid nesting:
```suggestion
		if !d.backwardsCompatible {
			panic("Restriction kind is not supported")
		}

		restrictionsValue := obj.Get(restrictionsKey)
		typeValue := obj.Get(typeKey)
		return d.decodeDeprecatedRestrictedType(
			typeValue,
			toSlice(restrictionsValue),
			results,
		)
```

Reply:
got it, updated